src/notify.c

```diff
@@ -388,8 +388,27 @@ void graceful_clear(void)
 
     control_unlock();
 }
+
 #endif /* SOUND_NOTIFY */
 
+/* Kills all notifications for `id`. This must be called before freeing a ToxWindow. */
+void kill_notifs(int id)
+{
+    control_lock();
+
+    for (size_t i = 0; i < ACTIVE_NOTIFS_MAX; ++i) {
+        if (actives[i].id_indicator == NULL) {
+            continue;
+        }
+
+        if (*actives[i].id_indicator == id) {
+            memset(&actives[i], 0, sizeof(struct _ActiveNotifications));
```

Comment:
When using memset to zero a struct w/ pointers in you should set the pointers to NULL after. actives[i].id_indicator,box aren't guaranteed to be NULL after this line.